### PR TITLE
Folder path, login page, separate service (#108, #122, #132)

### PR DESCRIPTION
**Pull request by AndrePu**
- Added a comfortable way to get main paths in clientApp.
**More detailed info:**
#108 Folder path

- Added login accordingly to a new component architecture rules.
**More detailed info:**
#110 Presentational and Container components

- Added authGuard on routings:
**More detailed info:**
#122 Common layout and routings

- Moved logic responsible for authentification from user.service to a new one.
**More detailed info:**
#132 Create separate service for login

### CODE REVIEW

File: Cooper/ClientApp/src.v2/app/app-routing.module.ts [hunk 1]
```diff
@@ -1,13 +1,14 @@
 import { NgModule } from '@angular/core';
 import { Routes, RouterModule } from '@angular/router';
+import {AuthGuard} from './guards';
```
Comment (horba) [inline Cooper/ClientApp/src.v2/app/app-routing.module.ts]: "@services"

File: Cooper/ClientApp/src.v2/app/services/model-services/user.service.ts [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+@Injectable({
+  providedIn: 'root'
+})
+export class UserService {
+
+  constructor(private fb: FormBuilder, private router: Router, private http: HttpClient, private socialAuthService: AuthService) { }
+
+  public user: any;
+  readonly registrationUrl = '/registration';
+  readonly authUrl = '/auth/login';
```
Comment (horba) [inline Cooper/ClientApp/src.v2/app/services/model-services/user.service.ts]: const?

File: Cooper/ClientApp/src.v2/app/services/model-services/user.service.ts [hunk 3]
```diff
@@ -0,0 +1,119 @@
...
+  }
+
+  public socialSignIn(socialPlatform: string) {
+    this.socialAuthService.signIn(this.getProvider[socialPlatform]).then(
+      (userData) => {
+        if (socialPlatform == 'google') {
+          userData.id = userData.email;
+          userData.token = userData.idToken;
+        }
+        this.socialsignin(userData);
```
Comment (horba) [inline Cooper/ClientApp/src.v2/app/services/model-services/user.service.ts]: socialSignIn()

File: Cooper/ClientApp/src.v2/app/services/model-services/user.service.ts [hunk 4]
```diff
@@ -0,0 +1,119 @@
...
+      }
+    );
+  }
+
+  socialsignin(userData) {
+    var body = this.bodyCreator(userData, true);
```
Comment (horba) [inline Cooper/ClientApp/src.v2/app/services/model-services/user.service.ts]: let

File: Cooper/ClientApp/src.v2/app/services/model-services/user.service.ts [hunk 5]
```diff
@@ -0,0 +1,119 @@
...
+    this.router.navigate(['/myPage', 'my']);
+  }
+
+  public BadLogin() {
+    this.router.navigate(['/myPage', 'my']);
+  }
+
+  bodyCreator(userData, login) {
+    let result;
+    if (login === true)
```
Comment (horba) [inline Cooper/ClientApp/src.v2/app/services/model-services/user.service.ts]: if(login)

Comment (horba) [inline Cooper/ClientApp/src.v2/app/services/model-services/user.service.ts]: for bool variables more clear name such as **Is**Authorized or other adjective

File: Cooper/ClientApp/tsconfig.json [hunk 6]
```diff
@@ -17,6 +17,16 @@
     "lib": [
       "es2018",
       "dom"
-    ]
+    ],
+    "paths": {
+    "@app*": ["src.v2/app/*"],
+    "@components*": ["src.v2/app/components/*"],
```
Comment (horba) [inline Cooper/ClientApp/tsconfig.json]: "@components/*": ["src.v2/app/components/*"]

and for others the same

then you can import only from "@components" without index

If you'll have another problems, please, write me

File: Cooper/ClientApp/src.v2/app/layouts/login/login.layout.ts [hunk 7]
```diff
@@ -1,4 +1,11 @@
-import { Component, OnInit } from '@angular/core';
+import { Component, OnInit, ViewEncapsulation, Input } from '@angular/core';
+import { AuthentificationService } from '@services/authentification/authentification.service';
```
Comment (horba) [inline Cooper/ClientApp/src.v2/app/layouts/login/login.layout.ts]: from '@services"

File: Cooper/ClientApp/src.v2/app/services/authentification/authentification.service.ts [hunk 8]
```diff
@@ -0,0 +1,118 @@
...
+  readonly authUrl = '/auth/login';
+  readonly getProvider = {
+    'google': GoogleLoginProvider.PROVIDER_ID,
+    'facebook': FacebookLoginProvider.PROVIDER_ID
+  };
+
+  constructor(private router: Router, private http: HttpClient, private socialAuthService: AuthService) { }
+
+  public checkAuthentification(): void {
+    const Token: string = localStorage.getItem('JwtCooper');
```
Comment (horba) [inline Cooper/ClientApp/src.v2/app/services/authentification/authentification.service.ts]: upper letter?

File: Cooper/ClientApp/src.v2/app/services/authentification/authentification.service.ts [hunk 9]
```diff
@@ -0,0 +1,118 @@
...
+    this.router.navigate(['/platform/profile']);
+  }
+
+  public BadLogin() {
+    this.router.navigate(['/login', {failedLogin: true}]);
+  }
+
+  bodyCreator(userData, login) {
+    let result;
+    if (login === true)
```
Comment (horba) [inline Cooper/ClientApp/src.v2/app/services/authentification/authentification.service.ts]: naming

File: Cooper/ClientApp/src.v2/app/services/model-services/user.service.ts [hunk 10]
```diff
@@ -0,0 +1,11 @@
+import { Injectable } from '@angular/core';
+
+@Injectable({
+  providedIn: 'root'
+})
+export class UserService {
+
+  constructor() { }
+
+  public user: any;
```
Comment (horba) [inline Cooper/ClientApp/src.v2/app/services/model-services/user.service.ts]: I hope you add user.model User and use it

Comment (AndrePu) [reply Cooper/ClientApp/src.v2/app/services/model-services/user.service.ts]: I will when I'll be doing profile page

Comment (horba) [reply Cooper/ClientApp/src.v2/app/services/model-services/user.service.ts]: no problems)

File: Cooper/ClientApp/src.v2/app/services/authentification/authentification.service.ts [hunk 11]
```diff
@@ -0,0 +1,118 @@
...
+      }
+    );
+  }
+
+  private transferSocialDataToServer(userData: any) {
+    let body = this.createBody(userData, true);
+
+    this.http.post(this.authUrl, body).subscribe(
+      response => {
+        this.loginOK((response as any).token);
```
Comment (horba) [inline Cooper/ClientApp/src.v2/app/services/authentification/authentification.service.ts]: any?

File: Cooper/ClientApp/src.v2/app/services/authentification/authentification.service.ts [hunk 12]
```diff
@@ -0,0 +1,118 @@
...
+    }
+  }
+
+  public signIn(credentials: string): void {
+    this.http.post(this.authUrl, credentials, {
+      headers: new HttpHeaders({
+        'Content-Type': 'application/json'
+      })
+    }).subscribe(response => {
+      this.loginOK((response as any).token);
```
Comment (horba) [inline Cooper/ClientApp/src.v2/app/services/authentification/authentification.service.ts]: any

Comment (horba) [reply Cooper/ClientApp/src.v2/app/services/authentification/authentification.service.ts]: loginOk -> onSuccesLogin/succesAuthHandler
BadLogin -> onFailedLogin/failedAuthHandler

it isn't clear responsibility of these functions

Comment (horba) [reply Cooper/ClientApp/src.v2/app/services/authentification/authentification.service.ts]: Now BadLogin() looks like redirectToLoginPage()

File: Cooper/ClientApp/src.v2/app/services/authentification/authentification.service.ts [hunk 13]
```diff
@@ -0,0 +1,118 @@
...
+import { HttpClient, HttpHeaders } from '@angular/common/http';
+import { Router } from '@angular/router';
+import { AuthService, FacebookLoginProvider, GoogleLoginProvider } from 'ng-dynami-social-login';
+
+@Injectable({
+  providedIn: 'root'
+})
+export class AuthentificationService {
+
+  readonly registrationUrl;
```
Comment (horba) [inline Cooper/ClientApp/src.v2/app/services/authentification/authentification.service.ts]: const

File: Cooper/ClientApp/src.v2/app/services/authentification/authentification.service.ts [hunk 14]
```diff
@@ -0,0 +1,118 @@
...
+  readonly authUrl;
```
Comment (horba) [inline Cooper/ClientApp/src.v2/app/services/authentification/authentification.service.ts]: const

File: Cooper/ClientApp/src.v2/app/services/authentification/authentification.service.ts [hunk 15]
```diff
@@ -0,0 +1,118 @@
...
+    this.socialProvider = {
+      'google': GoogleLoginProvider.PROVIDER_ID,
+      'facebook': FacebookLoginProvider.PROVIDER_ID
+    };
+   }
+
+  public checkAuthentification(): void {
+    const token: string = localStorage.getItem('JwtCooper');
+    if (token) {
+      this.router.navigate(['/platform/profile']);
```
Comment (horba) [inline Cooper/ClientApp/src.v2/app/services/authentification/authentification.service.ts]: should redirect to home page

File: Cooper/ClientApp/src.v2/app/components/login/login.component.html [hunk 16]
```diff
@@ -1,3 +1,31 @@
...
-  login works!
-</p>
+<h1 class="h1--color h1--positioning">Sign In</h1>
+
+<form class="form-signin" #loginForm="ngForm" (ngSubmit)="signInClicked(loginForm)">
+
+  <div class="personal-data-form">
+
+    <mat-form-field class="mat-form-field">
+      <input matInput type="email" [(ngModel)]="username" name="Username" placeholder="Enter your nickname" />
```
Comment (horba) [inline Cooper/ClientApp/src.v2/app/components/login/login.component.html]: upper letter

File: Cooper/ClientApp/src.v2/app/components/login/login.component.html [hunk 17]
```diff
@@ -1,3 +1,31 @@
...
+    </mat-form-field>
+
+    <mat-form-field class="mat-form-field">
+      <input matInput type="password" [(ngModel)]="password" name="Password" placeholder="Enter your password"/>
```
Comment (horba) [inline Cooper/ClientApp/src.v2/app/components/login/login.component.html]: upper letter

File: Cooper/ClientApp/src.v2/app/components/login/login.component.html [hunk 18]
```diff
@@ -1,3 +1,31 @@
...
+
+<div class="text-help">
+  <p>
+    also you can <a routerLink="../registration" routerLinkActive="active">Sign up</a>
+    <br>or sign in by:
+  </p>
+</div>
+
+<div class="socials-container">
+  <img src="assets/imageKeeper/google-icon.png" class="social-icon" (click)="socialSignInClicked('google')" />
```
[the remark this answers is not in the document]
Comment (AndrePu) [reply Cooper/ClientApp/src.v2/app/components/login/login.component.html]: Got it

File: Cooper/ClientApp/src.v2/app/services/index.ts [hunk 19]
```diff
@@ -0,0 +1,4 @@
+
+export {AuthentificationService} from './authentification/authentification.service';
+
+export {UserService} from './model-services/user.service';
```
Comment (horba) [inline Cooper/ClientApp/src.v2/app/services/index.ts]: Actually, you don't need for each service folder

Comment (AndrePu) [reply Cooper/ClientApp/src.v2/app/services/index.ts]: This is comfortable way to operate services through folders. We have a lot of models on server side. The time will come when we need to add appropriate services to our client. It's comfortable way to group these services. For each another service we have two files. In order to make our orientation easier we group two relative files in one folder.

File: Cooper/ClientApp/src.v2/app/services/authentification/authentification.service.ts [hunk 20]
```diff
@@ -0,0 +1,122 @@
...
+
+export class AuthentificationService {
+
+  constructor(private router: Router, private http: HttpClient, private socialAuthService: AuthService) {
+   }
+
+  public checkAuthentification(): void {
+    const token: string = localStorage.getItem('JwtCooper');
+    if (token) {
+      this.router.navigate(['/platform/home']);
```
Comment (horba) [inline Cooper/ClientApp/src.v2/app/services/authentification/authentification.service.ts]: why function with name check changes state of program?

Comment (AndrePu) [reply Cooper/ClientApp/src.v2/app/services/authentification/authentification.service.ts]: Because I'm the one who calls himself ninja))

**Pull request merged**